shm/island.cpp

```diff
@@ -0,0 +1,18 @@
+#include "island.hpp"
+#include "coordinates.hpp"
+
+Island::Island(Coordinates position, Time* time) : position_(position), time_(time) {
+    store_ = std::make_shared<Store>(time);
```

Comment:
You can't substitute store here. If you pass store in constructor, You will be able to pass mock class in tests.